.github/workflows/changelog-generation.yml

```diff
@@ -26,14 +26,27 @@ jobs:
         run: |
             ./generate-changelog.sh
             cat CHANGELOG.md
-           
+
+      - name: Read Tag Name from VERSION File
+        id: tag_name
+        run: |
+          echo "tag_name=$(cat VERSION)" >> $GITHUB_ENV
+
+      - name: Get Base SHA from Tag
+        id: base_sha
+        run: |
+          echo "base_sha=$(git rev-list -n 1 "v${{ env.tag_name }}")" >> $GITHUB_ENV
+        continue-on-error: true
+
       - name: Create Pull Request
         uses: peter-evans/create-pull-request@v6
         with:
-          commit-message: "chore: autogenerate changelog"
-          title: "Changelog"
-          body: This PR contains an autogenerated changelog for newer commits. The new changelog can be modified [here](https://github.com/prometheus/client_golang/blob/changelog-branch/CHANGELOG.md).
+          commit-message: "chore: cut ${{ github.ref_name }}"
```

Comment:
`v${{ env.tag_name }}` contains the previous release version, which is `v1.19.0` in our case.

To get version for current release, I'm taking it from the branch name. So, if the release branch contains the patch, it will be reflected in the title and commit message.


For example in below screenshot, the release branch is named `release-1.21.1`, and this is how the autogenerated PR looks like:

![image](https://github.com/prometheus/client_golang/assets/75629410/5ade11c0-22e8-4947-9bb9-0a903502888a)

Reply:
Should I write a script to append the patch version if it is missing in the release branch name?

Reply:
In our [release process](https://github.com/prometheus/client_golang?tab=readme-ov-file#for-maintainers-release-process), we bump the version file before creating a new release. This means that the release branch will have the appropriate version number in the `VERSION` file